### PR TITLE
DL - OpenID Connect mock

### PR DESCRIPTION
**Pull request by douglasrlee**

### CODE REVIEW

File: strategy.js [hunk 1]
```diff
@@ -42,8 +45,10 @@ module.exports = function (passport, util) {
           } else {
             this.verify(access_token, refresh_token, token_response, profile, verified);
           };
-        } else { // arity === 4
+        } else if (arity === 4) {
           this.verify(access_token, refresh_token, profile, verified);
+        } else if (arity === 2) {
```
Comment (douglasrlee) [inline strategy.js]: This is the new thing.

File: test/strategy.test.js [hunk 2]
```diff
@@ -96,62 +96,99 @@ describe('#authenticate', function (){
       });
     });
 
-    context('when passReqToCallback is false', function () {
-      it('handles a verify method that asks for accessToken, refreshToken, and profile correctly', function (done) {
-        strategy = new Strategy({ callbackURL: '/cb' }, function (access_token, refresh_token, profile, cb) {
+    context('when the verify arity is 6', function () {
```
Comment (douglasrlee) [inline test/strategy.test.js]: This is just moving.

File: test/strategy.test.js [hunk 3]
```diff
@@ -96,62 +96,99 @@ describe('#authenticate', function (){
...
+    });
 
-      it('handles a verify method that asks for access token, refresh token, token response, and profile', function (done) {
-        strategy = new Strategy({ callbackURL: '/cb' }, function (access_token, refresh_token, token_response, profile, cb) {
-          cb(null, {
-            profile: profile,
-            token_response: token_response,
-            access_token: access_token,
-            refresh_token: refresh_token
+    context('when the verify arity is 5', function () {
```
Comment (douglasrlee) [inline test/strategy.test.js]: This is just moving.

File: test/strategy.test.js [hunk 4]
```diff
@@ -96,62 +96,99 @@ describe('#authenticate', function (){
...
+        });
 
-        strategy.authenticate(req, {});
       });
     });
 
-    context('when passReqToCallback is true', function () {
-      it('handles a verify method that asks for the request, access_token, refresh_token, and profile', function (done) {
-        strategy = new Strategy({ callbackURL: '/cb', passReqToCallback: true }, function (request, access_token, refresh_token, profile, cb) {
+    context('when the verify arity is 4', function () {
```
Comment (douglasrlee) [inline test/strategy.test.js]: This just moving around.

File: test/strategy.test.js [hunk 5]
```diff
@@ -166,38 +203,25 @@ describe('#authenticate', function (){
...
-      it('handles a verify method that asks for request, access token, refresh token, token response, and profile', function (done) {
-        strategy = new Strategy({ callbackURL: '/cb', passReqToCallback: true }, function (request, access_token, refresh_token, token_response, profile, cb) {
-          cb(null, {
-            request: request,
-            profile: profile,
-            token_response: token_response,
-            access_token: access_token,
-            refresh_token: refresh_token
-          });
+    context('when the verify arity is 2', function () {
```
Comment (douglasrlee) [inline test/strategy.test.js]: This is the new piece.

File: strategy.js [hunk 6]
```diff
@@ -42,8 +45,10 @@ module.exports = function (passport, util) {
...
+          this.verify(token_set, verified);
```
Comment (kylessnell) [inline strategy.js]: does actual passport allow this too?

Comment (douglasrlee) [inline strategy.js]: Yes, this is what the openid connect client uses.

**Pull request closed without merging**